packages/wonder-blocks-dropdown/src/components/combobox-live-region.tsx

```diff
@@ -0,0 +1,199 @@
+
+type Props = {
+    /**
+     * The index of the focused item in the listbox.
+     */
+    focusedIndex: number;
+    /**
+     * The index of the focused item in the multi-select combobox.
+     */
+    focusedMultiSelectIndex: number;
+    /**
+     * The labels associated with the live region.
+     */
+    labels?: Pick<
+        ComboboxLabels,
+        | "closedState"
+        | "liveRegionCurrentItem"
+        | "liveRegionListboxTotal"
+        | "liveRegionMultipleSelectionTotal"
+    >;
+
+    /**
+     * Whether the listbox is open/expanded.
+     */
+    opened?: boolean;
+    /**
+     * The list of items in the listbox.
+     */
+    options: Array<OptionItemComponent>;
+    /**
+     * The labels of the selected items (multi-select only).
+     */
+    multiSelectLabels: Array<string>;
+    /**
+     * The value of the selected item(s).
+     */
+    selected: MaybeValueOrValues;
+    /**
+     * Whether the use can select more than one option item. Defaults to
+     * `single`.
+     */
+    selectionType?: "single" | "multiple";
+    /**
+     * The testId used for the live region.
+     */
+    testId?: string;
+};
+
+/**
+ * A component that announces focus changes to Screen Readers.
+ *
+ * This is useful as there are lots of issues with `role="combobox"` + Safari
+ * when the browser is not capable of announcing correctly the status of the
+ * currently focused option item.
+ *
+ * @see https://bugs.webkit.org/show_bug.cgi?id=167671
+ */
+export function ComboboxLiveRegion({
+    focusedIndex,
+    focusedMultiSelectIndex,
+    labels = {
+        closedState: defaultComboboxLabels.closedState,
+        liveRegionCurrentItem: defaultComboboxLabels.liveRegionCurrentItem,
+        liveRegionListboxTotal: defaultComboboxLabels.liveRegionListboxTotal,
+        liveRegionMultipleSelectionTotal:
+            defaultComboboxLabels.liveRegionMultipleSelectionTotal,
+    },
+    multiSelectLabels,
+    opened,
+    options,
+    selected,
+    selectionType = "single",
+    testId,
+}: Props) {
+    const lastSelectedValue = React.useRef<MaybeValueOrValues>(null);
+    const [message, setMessage] = React.useState("");
+
+    // Announce when an item is selected.
+    React.useEffect(() => {
+        if (selected !== lastSelectedValue?.current) {
+            let newMessage = "";
+            const lastSelectedLength = lastSelectedValue?.current?.length ?? 0;
+            const selectedLength = selected?.length ?? 0;
+            const selectedState =
+                selectedLength > lastSelectedLength
+                    ? "selected"
+                    : "not selected";
+            // Multi-select combobox.
+            if (Array.isArray(selected) && selected.length > 0) {
+                newMessage = `${multiSelectLabels.join(", ")} ${selectedState}`;
+            } else {
+                // Announces the selected item for single select combobox.
+                newMessage = multiSelectLabels[0] + " selected";
```

Comment:
Should we get the selected label from the options and selected props? From the prop description for `multiSelectLabels`, it seems like it should only be used for the multiselect!